### PR TITLE
flow typings for json-source-map

### PR DESCRIPTION
**Pull request by gorakong**
**Test Plan**:
- verified that no flow errors are present

Note: Removed `nullthrows` from `packages/core/diagnostic/src/diagnostic.js` since these typings should guard against variable `pos` (line 216) from being null.

### CODE REVIEW

**Comment by parcel-benchmark** [comment]
## Benchmark Results
<details><summary>Kitchen Sink ✅</summary><p>

#### Timings

| Description | Time | Difference |
| --- | --- | --- |
| Cold | 6.94s | -159.00ms |
| Cached | 1.63s | +9.00ms |

#### Cold Bundles

| Bundle | Size | Difference | Time | Difference |
| --- | --- | --- | --- | --- |
| dist/legacy/parcel.d5807e82.webp | 102.94kb | +0.00b | 217.00ms | -20.00ms 🚀 |

#### Cached Bundles

| Bundle | Size | Difference | Time | Difference |
| --- | --- | --- | --- | --- |
| dist/legacy/index.45cccf70.js | 1.10kb | +0.00b | 1.34s | -72.00ms 🚀 |
| dist/modern/index.html | 701.00b | +0.00b | 1.29s | -99.00ms 🚀 |

</p></details>

<details><summary>React HackerNews ✅</summary><p>

#### Timings

| Description | Time | Difference |
| --- | --- | --- |
| Cold | 25.44s | -1.79s 🚀 |
| Cached | 1.61s | -12.00ms |

#### Cold Bundles

| Bundle | Size | Difference | Time | Difference |
| --- | --- | --- | --- | --- |
| dist/index.js | 478.50kb | +0.00b | 10.61s | -626.00ms 🚀 |
| dist/PermalinkedComment.254739e9.js | 4.22kb | +0.00b | 5.53s | -386.00ms 🚀 |
| dist/UserProfile.bb46ff21.js | 1.70kb | +0.00b | 5.53s | -386.00ms 🚀 |
| dist/NotFound.9af20955.js | 532.00b | +0.00b | 776.00ms | -295.00ms 🚀 |
| dist/logo.24c8bf9e.png | 274.00b | +0.00b | 143.00ms | -24.00ms 🚀 |

#### Cached Bundles

*No bundle changes detected.*

</p></details>

<details><summary>AtlasKit Editor ✅</summary><p>

#### Timings

| Description | Time | Difference |
| --- | --- | --- |
| Cold | 2.91m | -4.53s |
| Cached | 4.58s | -216.00ms |

#### Cold Bundles

| Bundle | Size | Difference | Time | Difference |
| --- | --- | --- | --- | --- |
| dist/pdfRenderer.f8165fbc.js | 1.11mb | +0.00b | 50.11s | -3.24s 🚀 |
| dist/EmojiPickerComponent.ab132fac.js | 75.00kb | +0.00b | 10.45s | -1.60s 🚀 |
| dist/media-viewer.4c3f658c.js | 71.88kb | +0.00b | 14.37s | -2.32s 🚀 |
| dist/card.e1c0f362.js | 54.82kb | +0.00b | 12.25s | -1.81s 🚀 |
| dist/card.9d364a71.js | 52.57kb | +0.00b | 54.44s | -43.23s 🚀 |
| dist/Modal.f7e71087.js | 41.31kb | +0.00b | 5.36s | -1.70s 🚀 |
| dist/component.327022bf.js | 30.85kb | +0.00b | 6.52s | +664.00ms ⚠️ |
| dist/component.49a28949.js | 22.36kb | +0.00b | 5.36s | -1.70s 🚀 |
| dist/EmojiPickerComponent.7963a811.js | 18.12kb | +0.00b | 10.45s | -1.60s 🚀 |
| dist/dropzone.498ec003.js | 17.30kb | +0.00b | 53.98s | -43.69s 🚀 |
| dist/js.eb39c3cf.js | 16.55kb | +0.00b | 5.36s | -1.70s 🚀 |
| dist/ui.cfe8a82a.js | 13.94kb | +0.00b | 12.25s | -1.81s 🚀 |
| dist/workerHasher.9dc1181e.js | 11.90kb | +0.00b | 12.25s | -1.06s 🚀 |
| dist/component.fe59914e.js | 6.19kb | +0.00b | 5.85s | -3.54s 🚀 |
| dist/card.7b75c38c.js | 5.67kb | +0.00b | 14.36s | -2.32s 🚀 |
| dist/media-viewer.39298996.js | 4.01kb | +0.00b | 14.36s | -2.39s 🚀 |
| dist/EmojiPickerComponent.0c594241.js | 3.68kb | +0.00b | 9.97s | -2.08s 🚀 |
| dist/png-chunks-extract.3bcd3532.js | 3.55kb | +0.00b | 5.36s | -1.70s 🚀 |
| dist/dropzone.cca9436f.js | 3.32kb | +0.00b | 50.11s | -3.38s 🚀 |
| dist/Modal.725e4b5d.js | 3.07kb | +0.00b | 5.36s | -1.70s 🚀 |
| dist/16.984a552d.js | 2.49kb | +0.00b | 1.56s | -2.08s 🚀 |
| dist/card.bc6d3fe8.js | 2.15kb | +0.00b | 50.11s | +19.27s ⚠️ |
| dist/ResourcedEmojiComponent.4e7708b7.js | 2.07kb | +0.00b | 6.38s | +562.00ms ⚠️ |
| dist/date.1de12427.js | 1.88kb | +0.00b | 6.44s | -1.43s 🚀 |
| dist/16.51a973ae.js | 1.86kb | +0.00b | 6.24s | -1.31s 🚀 |
| dist/images.bd2c65c2.js | 1.82kb | +0.00b | 6.58s | -1.35s 🚀 |
| dist/16.664f2364.js | 1.79kb | +0.00b | 5.85s | -1.47s 🚀 |
| dist/feedback.762993d4.js | 1.78kb | +0.00b | 10.45s | +1.49s ⚠️ |
| dist/workerHasher.c6b3a06a.js | 1.67kb | +0.00b | 1.59m | +41.49s ⚠️ |
| dist/list-number.140c0835.js | 1.60kb | +0.00b | 6.58s | -1.48s 🚀 |
| dist/status.eb990fdd.js | 1.60kb | +0.00b | 6.79s | -1.73s 🚀 |
| dist/code.3d01c3f6.js | 1.52kb | +0.00b | 6.43s | -1.36s 🚀 |
| dist/16.d03ef2ed.js | 1.51kb | +0.00b | 6.04s | -1.29s 🚀 |
| dist/16.2560b363.js | 1.46kb | +0.00b | 5.85s | -3.68s 🚀 |
| dist/16.3d2fa847.js | 1.46kb | +0.00b | 6.04s | -1.29s 🚀 |
| dist/16.f8607434.js | 1.45kb | +0.00b | 1.56s | -2.08s 🚀 |
| dist/link.0f2a27b0.js | 1.44kb | +0.00b | 6.58s | -1.44s 🚀 |
| dist/heading6.e4e9d5d4.js | 1.44kb | +0.00b | 6.90s | -2.04s 🚀 |
| dist/16.a99d351e.js | 1.44kb | +0.00b | 6.43s | -1.35s 🚀 |
| dist/heading3.94ce1091.js | 1.43kb | +0.00b | 6.90s | -1.97s 🚀 |
| dist/16.c8c5e38d.js | 1.40kb | +0.00b | 6.24s | -1.54s 🚀 |
| dist/emoji.ea46e5f5.js | 1.37kb | +0.00b | 6.58s | -1.34s 🚀 |
| dist/16.039b21a1.js | 1.36kb | +0.00b | 6.24s | -1.31s 🚀 |
| dist/16.f10fd24d.js | 1.34kb | +0.00b | 1.56s | -2.08s 🚀 |
| dist/16.8938ddc0.js | 1.32kb | +0.00b | 6.24s | -1.31s 🚀 |
| dist/heading5.87b6dae6.js | 1.31kb | +0.00b | 6.90s | -2.04s 🚀 |
| dist/16.955db3fc.js | 1.31kb | +0.00b | 6.43s | -1.36s 🚀 |
| dist/16.d58d6149.js | 1.27kb | +0.00b | 1.56s | -2.08s 🚀 |
| dist/16.3d8988cf.js | 1.27kb | +0.00b | 6.04s | -1.51s 🚀 |
| dist/16.b709596e.js | 1.26kb | +0.00b | 5.85s | -1.47s 🚀 |
| dist/16.202504a1.js | 1.26kb | +0.00b | 6.04s | -1.29s 🚀 |
| dist/16.5ba30215.js | 1.26kb | +0.00b | 6.04s | -4.27s 🚀 |
| dist/16.2da93a26.js | 1.26kb | +0.00b | 5.85s | -3.54s 🚀 |
| dist/heading2.e0e7648d.js | 1.25kb | +0.00b | 6.90s | -1.78s 🚀 |
| dist/16.a2db69a4.js | 1.23kb | +0.00b | 1.56s | -2.08s 🚀 |
| dist/mention.41dd4317.js | 1.21kb | +0.00b | 6.67s | -1.38s 🚀 |
| dist/heading4.926b16d5.js | 1.20kb | +0.00b | 6.90s | -1.97s 🚀 |
| dist/layout.a7a50d60.js | 1.19kb | +0.00b | 6.58s | -1.44s 🚀 |
| dist/16.eb0e686d.js | 1.18kb | +0.00b | 6.24s | -1.31s 🚀 |
| dist/divider.6cc8acb0.js | 1.17kb | +0.00b | 8.93s | +1.01s ⚠️ |
| dist/quote.b58dc9d4.js | 1.16kb | +0.00b | 6.78s | -1.73s 🚀 |
| dist/component.f51f4183.js | 1.15kb | +0.00b | 6.38s | +562.00ms ⚠️ |
| dist/action.c1329039.js | 1.15kb | +0.00b | 6.43s | -1.36s 🚀 |
| dist/decision.b8aab464.js | 1.13kb | +0.00b | 8.93s | +1.06s ⚠️ |
| dist/panel-warning.a4dc1737.js | 1.13kb | +0.00b | 6.67s | -1.62s 🚀 |
| dist/list.6772c571.js | 1.10kb | +0.00b | 9.26s | +1.21s ⚠️ |
| dist/heading1.e7155c65.js | 1.09kb | +0.00b | 6.79s | -1.83s 🚀 |
| dist/panel-error.2d0a3dc5.js | 1.03kb | +0.00b | 6.67s | -1.54s 🚀 |
| dist/panel.983b8f36.js | 1.02kb | +0.00b | 6.79s | -1.51s 🚀 |
| dist/table.613566d1.js | 1.01kb | +0.00b | 6.79s | -1.86s 🚀 |
| dist/panel-success.f52c4b4f.js | 990.00b | +0.00b | 6.67s | -1.62s 🚀 |
| dist/panel-note.af0bab14.js | 986.00b | +0.00b | 6.67s | -1.54s 🚀 |
| dist/media-card-analytics-error-boundary.eaabcc04.js | 956.00b | +0.00b | 12.25s | -1.81s 🚀 |
| dist/media-card-analytics-error-boundary.a56d8c27.js | 956.00b | +0.00b | 1.59m | +41.50s ⚠️ |
| dist/media-viewer-analytics-error-boundary.6a26a5d6.js | 876.00b | +0.00b | 14.37s | -2.39s 🚀 |
| dist/simpleHasher.948d1b20.js | 673.00b | +0.00b | 12.25s | -1.06s 🚀 |
| dist/index.html | 119.00b | +0.00b | 7.55s | +460.00ms ⚠️ |

#### Cached Bundles

| Bundle | Size | Difference | Time | Difference |
| --- | --- | --- | --- | --- |
| dist/index.1a36c238.js | 2.71mb | +0.00b | 365.00ms | -37.00ms 🚀 |
| dist/pdfRenderer.f8165fbc.js | 1.11mb | +3.00b ⚠️ | 623.00ms | -64.00ms 🚀 |
| dist/editorView.8e3d6b32.js | 586.93kb | +0.00b | 667.00ms | -47.00ms 🚀 |
| dist/popup.95b6e35c.js | 169.09kb | +0.00b | 655.00ms | -49.00ms 🚀 |
| dist/Toolbar.b45f0082.js | 92.27kb | +0.00b | 667.00ms | -47.00ms 🚀 |
| dist/EmojiPickerComponent.ab132fac.js | 75.00kb | +0.00b | 595.00ms | -73.00ms 🚀 |
| dist/media-viewer.b113ec80.js | 71.78kb | +0.00b | 614.00ms | -73.00ms 🚀 |
| dist/card.e15cb2dd.js | 54.95kb | +0.00b | 614.00ms | -73.00ms 🚀 |
| dist/card.3dbd8c2c.js | 52.71kb | +0.00b | 635.00ms | -83.00ms 🚀 |
| dist/Modal.9a26da87.js | 41.08kb | +0.00b | 388.00ms | -28.00ms 🚀 |
| dist/component.327022bf.js | 30.85kb | +0.00b | 364.00ms | -37.00ms 🚀 |
| dist/component.49a28949.js | 22.36kb | +0.00b | 388.00ms | -28.00ms 🚀 |
| dist/DatePicker.1626269c.js | 20.94kb | +0.00b | 590.00ms | -66.00ms 🚀 |
| dist/EmojiPickerComponent.7963a811.js | 18.12kb | +0.00b | 595.00ms | -73.00ms 🚀 |
| dist/dropzone.cfa9b383.js | 17.22kb | +0.00b | 630.00ms | -74.00ms 🚀 |
| dist/smartMediaEditor.204c7bbc.js | 16.79kb | +0.00b | 638.00ms | -80.00ms 🚀 |
| dist/js.eb39c3cf.js | 16.55kb | +0.00b | 388.00ms | -28.00ms 🚀 |
| dist/ui.cfe8a82a.js | 13.94kb | +0.00b | 608.00ms | -69.00ms 🚀 |
| dist/workerHasher.9dc1181e.js | 11.90kb | +0.00b | 606.00ms | -71.00ms 🚀 |
| dist/EmojiTypeAheadComponent.8313d89f.js | 8.61kb | +0.00b | 606.00ms | -71.00ms 🚀 |
| dist/component.fe59914e.js | 6.19kb | +0.00b | 398.00ms | -27.00ms 🚀 |
| dist/card.7b75c38c.js | 5.67kb | +0.00b | 614.00ms | -73.00ms 🚀 |
| dist/media-viewer.39298996.js | 4.01kb | +0.00b | 655.00ms | -48.00ms 🚀 |
| dist/EmojiPickerComponent.d165b446.js | 3.56kb | +0.00b | 595.00ms | -73.00ms 🚀 |
| dist/png-chunks-extract.3bcd3532.js | 3.55kb | +0.00b | 388.00ms | -28.00ms 🚀 |
| dist/dropzone.cca9436f.js | 3.32kb | +0.00b | 626.00ms | -78.00ms 🚀 |
| dist/clipboard.dddd75a9.js | 2.83kb | +0.00b | 626.00ms | -78.00ms 🚀 |
| dist/16.984a552d.js | 2.49kb | +0.00b | 380.00ms | -27.00ms 🚀 |
| dist/EmojiUploadComponent.c023b9c0.js | 2.22kb | +0.00b | 599.00ms | -69.00ms 🚀 |
| dist/card.bc6d3fe8.js | 2.15kb | +0.00b | 655.00ms | -48.00ms 🚀 |
| dist/ResourcedEmojiComponent.4e7708b7.js | 2.07kb | +0.00b | 365.00ms | -37.00ms 🚀 |
| dist/date.1de12427.js | 1.88kb | +0.00b | 479.00ms | -58.00ms 🚀 |
| dist/16.51a973ae.js | 1.86kb | +0.00b | 425.00ms | -59.00ms 🚀 |
| dist/images.bd2c65c2.js | 1.82kb | +0.00b | 497.00ms | -59.00ms 🚀 |
| dist/16.664f2364.js | 1.79kb | +0.00b | 405.00ms | -61.00ms 🚀 |
| dist/feedback.762993d4.js | 1.78kb | +0.00b | 579.00ms | -76.00ms 🚀 |
| dist/browser.4322401e.js | 1.70kb | +0.00b | 626.00ms | -78.00ms 🚀 |
| dist/workerHasher.4a170deb.js | 1.67kb | +0.00b | 606.00ms | -71.00ms 🚀 |
| dist/workerHasher.c6b3a06a.js | 1.67kb | +0.00b | 635.00ms | -83.00ms 🚀 |
| dist/list-number.140c0835.js | 1.60kb | +0.00b | 509.00ms | -71.00ms 🚀 |
| dist/status.eb990fdd.js | 1.60kb | +0.00b | 550.00ms | -66.00ms 🚀 |
| dist/code.3d01c3f6.js | 1.52kb | +0.00b | 479.00ms | -58.00ms 🚀 |
| dist/16.d03ef2ed.js | 1.51kb | +0.00b | 420.00ms | -64.00ms 🚀 |
| dist/16.2560b363.js | 1.46kb | +0.00b | 405.00ms | -61.00ms 🚀 |
| dist/16.3d2fa847.js | 1.46kb | +0.00b | 417.00ms | -49.00ms 🚀 |
| dist/link.0f2a27b0.js | 1.44kb | +0.00b | 504.00ms | -52.00ms 🚀 |
| dist/heading6.e4e9d5d4.js | 1.44kb | +0.00b | 579.00ms | -76.00ms 🚀 |
| dist/16.a99d351e.js | 1.44kb | +0.00b | 456.00ms | -60.00ms 🚀 |
| dist/heading3.94ce1091.js | 1.43kb | +0.00b | 567.00ms | -89.00ms 🚀 |
| dist/16.c8c5e38d.js | 1.40kb | +0.00b | 443.00ms | -73.00ms 🚀 |
| dist/emoji.ea46e5f5.js | 1.37kb | +0.00b | 497.00ms | -59.00ms 🚀 |
| dist/16.039b21a1.js | 1.36kb | +0.00b | 443.00ms | -73.00ms 🚀 |
| dist/16.8938ddc0.js | 1.32kb | +0.00b | 443.00ms | -73.00ms 🚀 |
| dist/heading5.87b6dae6.js | 1.31kb | +0.00b | 579.00ms | -76.00ms 🚀 |
| dist/16.955db3fc.js | 1.31kb | +0.00b | 478.00ms | -59.00ms 🚀 |
| dist/expand.983a2c37.js | 1.30kb | +0.00b | 579.00ms | -77.00ms 🚀 |
| dist/16.3d8988cf.js | 1.27kb | +0.00b | 421.00ms | -63.00ms 🚀 |
| dist/16.b709596e.js | 1.26kb | +0.00b | 405.00ms | -61.00ms 🚀 |
| dist/16.202504a1.js | 1.26kb | +0.00b | 420.00ms | -63.00ms 🚀 |
| dist/16.5ba30215.js | 1.26kb | +0.00b | 420.00ms | -64.00ms 🚀 |
| dist/16.2da93a26.js | 1.26kb | +0.00b | 405.00ms | -61.00ms 🚀 |
| dist/media-card-analytics-error-boundary.7bdb6beb.js | 1.25kb | +0.00b | 655.00ms | -48.00ms 🚀 |
| dist/heading2.e0e7648d.js | 1.25kb | +0.00b | 567.00ms | -89.00ms 🚀 |
| dist/mention.41dd4317.js | 1.21kb | +0.00b | 510.00ms | -70.00ms 🚀 |
| dist/heading4.926b16d5.js | 1.20kb | +0.00b | 572.00ms | -84.00ms 🚀 |
| dist/layout.a7a50d60.js | 1.19kb | +0.00b | 497.00ms | -59.00ms 🚀 |
| dist/16.eb0e686d.js | 1.18kb | +0.00b | 443.00ms | -73.00ms 🚀 |
| dist/divider.6cc8acb0.js | 1.17kb | +0.00b | 497.00ms | -59.00ms 🚀 |
| dist/quote.b58dc9d4.js | 1.16kb | +0.00b | 519.00ms | -97.00ms 🚀 |
| dist/component.f51f4183.js | 1.15kb | +0.00b | 365.00ms | -37.00ms 🚀 |
| dist/action.c1329039.js | 1.15kb | +0.00b | 478.00ms | -59.00ms 🚀 |
| dist/decision.b8aab464.js | 1.13kb | +0.00b | 495.00ms | -42.00ms 🚀 |
| dist/panel-warning.a4dc1737.js | 1.13kb | +0.00b | 519.00ms | -97.00ms 🚀 |
| dist/list.6772c571.js | 1.10kb | +0.00b | 509.00ms | -71.00ms 🚀 |
| dist/heading1.e7155c65.js | 1.09kb | +0.00b | 567.00ms | -89.00ms 🚀 |
| dist/panel-error.2d0a3dc5.js | 1.03kb | +0.00b | 510.00ms | -70.00ms 🚀 |
| dist/panel.983b8f36.js | 1.02kb | +0.00b | 519.00ms | -97.00ms 🚀 |
| dist/table.613566d1.js | 1.01kb | +0.00b | 567.00ms | -89.00ms 🚀 |
| dist/panel-success.f52c4b4f.js | 990.00b | +0.00b | 518.00ms | -98.00ms 🚀 |
| dist/panel-note.af0bab14.js | 986.00b | +0.00b | 514.00ms | -66.00ms 🚀 |
| dist/media-picker-analytics-error-boundary.debe5415.js | 960.00b | +0.00b | 626.00ms | -78.00ms 🚀 |
| dist/media-card-analytics-error-boundary.eaabcc04.js | 956.00b | +0.00b | 614.00ms | -73.00ms 🚀 |
| dist/media-card-analytics-error-boundary.a56d8c27.js | 956.00b | +0.00b | 635.00ms | -83.00ms 🚀 |
| dist/media-viewer-analytics-error-boundary.6a26a5d6.js | 876.00b | +0.00b | 655.00ms | -48.00ms 🚀 |
| dist/EmojiPickerComponent.57313a93.js | 680.00b | +0.00b | 595.00ms | -73.00ms 🚀 |
| dist/simpleHasher.948d1b20.js | 673.00b | +0.00b | 607.00ms | -70.00ms 🚀 |
| dist/simpleHasher.2437bfc1.js | 673.00b | +0.00b | 635.00ms | -83.00ms 🚀 |

</p></details>

<details><summary>Three.js x4 🚨</summary><p>

#### Timings

| Description | Time | Difference |
| --- | --- | --- |
| Cold | **FAILED** | -0.00ms |
| Cached | **FAILED** | -0.00ms |

#### Cold Bundles

*No bundles found, this is probably a failed build...*
#### Cached Bundles

*No bundles found, this is probably a failed build...*
</p></details>

**[Click here to view a detailed benchmark overview.](https://benchmark.parceljs.org/benchmark/281952187455635976)**

File: flow-libs/json-source-map.js.flow [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    value: Position,
+    valueEnd: Position,
+    key?: Position,
+    keyEnd?: Position,
+  |};
+
+  declare export default {|
+    parse: (
+      json: string,
+      _?: any,
```
Comment (wbinnssmith) [inline flow-libs/json-source-map.js.flow:25]: `mixed` rather than `any`? same below

File: flow-libs/json-source-map.js.flow [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+      opts?: {|bigint: boolean|},
+    ) => {|
+      data: {...},
+      pointers: {[key: string]: Mapping, ...},
```
Comment (wbinnssmith) [inline flow-libs/json-source-map.js.flow:29]: can this be an exact type? `{|[key: string]: Mapping|}` same below

File: flow-libs/json-source-map.js.flow [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    keyEnd?: Position,
+  |};
+
+  declare export default {|
+    parse: (
+      json: string,
+      _?: any,
+      opts?: {|bigint: boolean|},
+    ) => {|
+      data: {...},
```
Comment (wbinnssmith) [inline flow-libs/json-source-map.js.flow:28]: should this be `mixed`?

File: packages/core/diagnostic/src/diagnostic.js [hunk 4]
```diff
@@ -213,7 +212,7 @@ export function generateJSONCodeHighlights(
   // json-source-map doesn't support a tabWidth option (yet)
   let map = jsonMap.parse(code.replace(/\t/g, ' '));
   return ids.map(({key, type, message}) => {
-    let pos = nullthrows(map.pointers[key]);
+    let pos = map.pointers[key];
```
Comment (mischnic) [inline packages/core/diagnostic/src/diagnostic.js:215]: I'm afraid that Flow cannot always be trusted 😬 

`item` isn't a number in this case, but Flow doesn't complain.
```
type Pointers =  {|[key: string]: number|};

let x: Pointers = {b: 1};

let item = x['a'];
(item: number);
```

https://flow.org/try/#0C4TwDgpgBACg9gSwHbAgJwM5QLxSgbwB8BtAawhAC4oNg1kBzAXWqQFcBbAI3UIF8A3ACghAGwjAoAD2rxkqTDgKCR4yQlQclU4gHIAhrqbCAFBogdWnHmgCUAoA

In our situation here, the `pos` parameter of `getJSONSourceLocation` could suddenly be `undefined`, although Flow says it's an object.

Comment (gorakong) [reply packages/core/diagnostic/src/diagnostic.js]: @mischnic Good catch! I've added `nullthrows` back.

**Review by wbinnssmith — APPROVED** [review]

**Pull request merged**